src/Common/src/System/Dynamic/Utils/ExpressionUtils.cs

```diff
@@ -171,6 +171,11 @@ public static Expression ValidateOneArgument(MethodBase method, ExpressionType n
             return arguments;
         }
 
+        public static void RequiresCanRead(Expression expression, string paramName)
+        {
+            RequiresCanRead(expression, paramName, -1);
+        }
```

Comment:
Well, this is not much different from  passing the default "-1" and then checking for that same "-1" and ignoring it in downstream helpers.  
Is it possible to _not_ pass -1 and _not_ check for it  in helpers?

The point is that most common scenario is when the re is no index, so we should have a simpler codepath for those cases that does not involve the index at all.